pkg/csi_driver/controller.go

```diff
@@ -115,25 +103,46 @@ func (s *controllerServer) CreateVolume(ctx context.Context, req *csi.CreateVolu
 	if filer != nil {
 		// Instance already exists, check if it meets the request
 		if err = file.CompareInstances(newFiler, filer); err != nil {
-			s.config.ipAllocator.Unreserve(reservedIPRange)
 			return nil, status.Error(codes.AlreadyExists, err.Error())
 		}
 	} else {
+		// Get the reserved-IPV4-cidr from the request. It was not added in the service instance as it is only consumed by the controller
+		reservedIPV4CIDR, reservedIPV4CIDRPresent := req.GetParameters()[paramReservedIPV4CIDR]
+
+		if !reservedIPV4CIDRPresent {
+			return nil, fmt.Errorf("The required parameter reserved-ipv4-cidr was not provided")
```

Comment:
This is not a required parameter.  If it is not specified, then we leave the ReservedIPRange field blank and let Filestore choose an available IP